crates/video/src/video.rs

```diff
@@ -0,0 +1,1125 @@
+    padding: u32,
+    pub tag: [[u8; 32]; 2],
+}
+
+impl ToBytes for VideoStreamCreateRequest {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut ret: Vec<u8> = Vec::new();
+
+        ret.extend_from_slice(self.hdr.as_slice());
+        ret.extend_from_slice(self.in_mem_type.as_slice());
+        ret.extend_from_slice(self.out_mem_type.as_slice());
+        ret.extend_from_slice(self.coded_format.as_slice());
+        ret.extend_from_slice(self.padding.as_slice());
+        ret.extend_from_slice(self.tag[0].as_slice());
+        ret.extend_from_slice(self.tag[1].as_slice());
+
+        ret
+    }
+}
+
+// SAFETY: The layout of the structure is fixed and can be initialized by
+// reading its content from byte array.
+unsafe impl ByteValued for VideoStreamCreateRequest {}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct VideoResourceCreateRequest {
+    pub hdr: VirtioVideoHeader,
+    pub queue_type: Le32,
+    pub resource_id: Le32,
+    pub planes_layout: Le32,
+    pub num_planes: Le32,
+    pub plane_offsets: [Le32; VIRTIO_VIDEO_MAX_PLANES as usize],
+    pub num_entries: [Le32; VIRTIO_VIDEO_MAX_PLANES as usize],
+}
+
+// SAFETY: The layout of the structure is fixed and can be initialized by
+// reading its content from byte array.
+unsafe impl ByteValued for VideoResourceCreateRequest {}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct VideoResourceQueueRequest {
+    pub hdr: VirtioVideoHeader,
+    pub queue_type: Le32,
+    pub resource_id: Le32,
+    pub timestamp: Le64,
+    pub num_data_sizes: Le32,
+    pub data_sizes: [Le32; VIRTIO_VIDEO_MAX_PLANES as usize],
+    pub padding: Le32,
+}
+
+// SAFETY: The layout of the structure is fixed and can be initialized by
+// reading its content from byte array.
+unsafe impl ByteValued for VideoResourceQueueRequest {}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct VideoGetParamsRequest {
+    pub hdr: VirtioVideoHeader,
+    pub queue_type: Le32,
+    pub padding: Le32,
+}
+
+// SAFETY: The layout of the structure is fixed and can be initialized by
+// reading its content from byte array.
+unsafe impl ByteValued for VideoGetParamsRequest {}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct VideoSetParamsRequest {
+    pub hdr: VirtioVideoHeader,
+    pub params: virtio_video_params,
+}
+
+// SAFETY: The layout of the structure is fixed and can be initialized by
+// reading its content from byte array.
+unsafe impl ByteValued for VideoSetParamsRequest {}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) struct VideoGetControlRequest {
+    pub hdr: VirtioVideoHeader,
+    pub control: Le32,
+    pub padding: Le32,
+}
+
+// SAFETY: The layout of the structure is fixed and can be initialized by
+// reading its content from byte array.
+unsafe impl ByteValued for VideoGetControlRequest {}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct virtio_video_format_range {
```

Comment:
Very strange name of the struct :)

Reply:
I was translating driver struct names to CamelCase, and then at some point decided that I should keep them consistent with the kernel names. But ended up with inconsistent naming with my own crate :facepalm: 

I will change them to be all CamelCase.

Reply:
Eventually, once the kernel headers land, we would prefer to use them (through virtio-bindings). So I am not seeing an issue with keeping them kernel-style. But I got no strong feelings on this.

Reply:
Me neither, if we will end up with kernel-style anyway, I can already change them all to follow that. I was mostly agreeing in keeping consistent strategy for this crate.